haystack_experimental/components/writers/chat_message_writer.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import importlib
+from typing import Any, Dict, List
+
+from haystack import DeserializationError, component, default_from_dict, default_to_dict, logging
+from haystack.dataclasses import ChatMessage
+
+from haystack_experimental.chat_message_stores.types import ChatMessageStore
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class ChatMessageWriter:
+    """
+    Writes chat messages to an underlying ChatMessageStore.
+
+    Usage example:
+    ```python
+    from haystack.dataclasses import ChatMessage
+    from haystack_experimental.components.writers import ChatMessageWriter
+    from haystack_experimental.chat_message_stores.in_memory import InMemoryChatMessageStore
+
+    messages = [
+        ChatMessage.from_assistant("Hello, how can I help you?"),
+        ChatMessage.from_user("I have a question about Python."),
+    ]
+    message_store = InMemoryChatMessageStore()
+    writer = ChatMessageWriter(message_store)
+    writer.run(messages)
+    ```
+    """
+
+    def __init__(self, message_store: ChatMessageStore):
+        """
+        Create a ChatMessageWriter component.
+
+        :param message_store:
+            The ChatMessageStore where the chat messages are to be written.
+        """
+        self.message_store = message_store
+
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {"message_store": type(self.message_store).__name__}
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(self, message_store=self.message_store.to_dict())
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "ChatMessageWriter":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            The dictionary to deserialize from.
+        :returns:
+            The deserialized component.
+
+        :raises DeserializationError:
+            If the message store is not properly specified in the serialization data or its type cannot be imported.
+        """
+        init_params = data.get("init_parameters", {})
+        if "message_store" not in init_params:
+            raise DeserializationError("Missing 'message_store' in serialization data")
+        if "type" not in init_params["message_store"]:
+            raise DeserializationError("Missing 'type' in message store's serialization data")
```

Comment:
Seems superfluous.

Reply:
Copied it from from [DocumentWriter](https://github.com/deepset-ai/haystack/blob/main/haystack/components/writers/document_writer.py#L76) 
Perhaps leave for now and rework all at the same time for consistency?